tests/test_distributed.py

```diff
@@ -44,34 +69,66 @@ def _distributed_train_model(self, opt):
 
         return (valid, test)
 
-    def tearDown(self):
-        # we need to de-initialize the distributed world, otherwise other
-        # tests will they're we're distributed when we're really not.
-        dist.destroy_process_group()
-
     def test_generator_distributed(self):
-        valid, test = self._distributed_train_model(
-            dict(
-                task='integration_tests:nocandidate',
-                model='transformer/generator',
-                optimizer='adamax',
-                learningrate=7e-3,
-                batchsize=32,
-                validation_every_n_epochs=5,
-                num_epochs=20,
-                n_layers=1,
-                n_heads=1,
-                ffn_size=32,
-                embedding_size=32,
-                beam_size=1,
-            )
-        )
+        valid, test = self._distributed_train_model(self._base_config)
 
         self.assertLessEqual(valid['ppl'], 1.20)
         self.assertGreaterEqual(valid['bleu-4'], 0.95)
         self.assertLessEqual(test['ppl'], 1.20)
         self.assertGreaterEqual(test['bleu-4'], 0.95)
 
+        # Tests that DialogData.get() is doing the right thing
+        # Ensure no duplication of examples among workers
+        # It would be 200 if each worker did all the examples
+        self.assertEqual(valid['exs'].value(), 100)
+        self.assertEqual(test['exs'].value(), 100)
+
+    def test_distributed_eval_max_exs(self):
+        config = copy.deepcopy(self._base_config)
+        config['validation_max_exs'] = 90
+        config['short_final_eval'] = True
+        valid, test = self._distributed_train_model(config)
+
+        # Tests that DialogData.get() is doing the right thing
+        # Ensure no duplication of examples among workers
+        # It would be 200 if each worker did all the examples
+        # Note: we decided that it was OK for the total count to be slightly off
+        # when using validation_max_exs and distributed.
+        # It's > 90 b/c there are two workers, told to do 45 each, & BatchWorld
+        # parley() does batchsize examples each time, so each worker will do 49
+        # examples.
+        self.assertEqual(valid['exs'].value(), 98)
```

Comment:
I'm torn between testing the implementation (which gets 98) and testing the overall correctness (we don't go too overboard in VME).

I think the thing to do is to leave a comment saying "In the future, if we fix VME, this assert should be changed to exactly 90" so future people can understand how test is "wrong", not their implementation.